cmd/juju/application/bundle.go

```diff
@@ -601,6 +608,19 @@ func (h *bundleHandler) addApplication(change *bundlechanges.AddApplicationChang
 		return errors.Annotatef(err, "cannot deploy application %q", p.Application)
 	}
 	h.writeAddedResources(resNames2IDs)
+
+	if h.bundleURL == nil {
+		return nil
+	}
+
+	tag := names.NewApplicationTag(p.Application).String()
+	result, err := h.api.SetAnnotation(map[string]map[string]string{tag: {"bundleURL": h.bundleURL.String()}})
+	if err == nil && len(result) > 0 {
+		err = result[0].Error
+	}
+	if err != nil {
+		logger.Debugf("error setting bundleURL annotation for %q: %s", p.Application, err)
```

Comment:
If we are just going to swallow the error rather than stopping, I think we should use ctx.Infof (on the cmd.Context object) rather than logger.Debug.

This means it will be written out to stderr and more likely to be noticed.